src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/VaultAdapter.cs

```diff
@@ -117,9 +117,12 @@ public IEnumerable<AzureBackupVault> GetVaultsInResourceGroup(string resourceGro
         /// </summary>
         /// <param name="resourceGroupName"></param>
         /// <param name="vaultName"></param>
-        public void DeleteVault(string resourceGroupName, string vaultName)
+        public bool DeleteVault(string resourceGroupName, string vaultName)
         {
-            AzureBackupVaultClient.Vault.DeleteAsync(resourceGroupName, vaultName, GetCustomRequestHeaders(), CmdletCancellationToken).Wait();
+            AzureBackupVaultGetResponse response = AzureBackupVaultClient.Vault.DeleteAsync(resourceGroupName, vaultName, GetCustomRequestHeaders(), CmdletCancellationToken).Result;
+            
+            // OneSDK will return only either OK or NoContent
+            return response.StatusCode == System.Net.HttpStatusCode.OK;
```

Comment:
How will you differentiate with case where exception will come due to any other reason like service is done, etc? Those which doesn't give NoContent?